core/controllers/library.py

```diff
@@ -239,11 +303,8 @@ def get(self):
 
         # If there is a category parameter, it should be in the following form:
         #     category=("Algebra" OR "Math")
```

Comment:
From this comment it seems that there are more rules for `category`, should we validate that too?

Reply:
What I mean is that the category string is not only enclosed in brackets but it can also contain OR inside (maybe even AND?), also the there are `"` inside, so `is_string_contained_within_parenthesis` should also validate all these other things.

Reply:
Maybe change the function name to `is_gae_search_query_string` (As it's a [query string pattern](https://cloud.google.com/appengine/docs/standard/python/search/query_strings) for the gae_search)

Reply:
@DubeySandeep I have changed name, please take a look.

Reply:
@vojtechjelinek -- Yes it is true  that ```is_gae_search_query_string``` may contains OR or AND inside it but consider the case when we only have one data for example ```language_code=("en")```. 
For this type of data current version of check is enough but, for multiple data  OR or AND is added in the string.
The point is OR/AND may or may not be present so I want to make this validator a generic one.
Looking forward for your suggestions.

Reply:
Hmm, I mean you can do at least some kind of regex validation for this. The full validation might also check if OR/AND are present if they are present correctly, but that is probably and overkill for this place.

Reply:
Hmm, okay. Regex might not be good for this. But you can still doe a check inside the `is_gae_search_query_string` not only for enclosing but also for allowing characters, or for the structure.

Also please rename `is_gae_search_query_string` to `is_search_query_string`

Reply:
Please help me to understand what changes you are looking for this method.

Reply:
Added extra validators for category and language code.